core-bundle/src/Messenger/EventListener/MessageListener.php

```diff
@@ -12,16 +12,31 @@
 
 namespace Contao\CoreBundle\Messenger\EventListener;
 
+use Doctrine\DBAL\Connection;
 use Psr\Log\LoggerInterface;
 use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
 use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
 
 class MessageListener
 {
-    public function __construct(private readonly LoggerInterface $logger)
+    public function __construct(private readonly LoggerInterface $logger, private readonly Connection $connection)
     {
     }
 
+    #[AsEventListener]
+    public function onWorkerRunning(WorkerRunningEvent $event): void
+    {
+        if (!$event->isWorkerIdle()) {
+            return;
+        }
+
+        // In case Doctrine is used as message broker, we want to close the connection when
+        // the worker is idle. Actually, this also seems useful when Doctrine is not used
+        // as message broker anyway to prevent dangling connections.
```

Comment:
I think we can summarise this to:

```suggestion
        // Close database connection when the worker is idle (#8199)
```